.github/workflows/CI-build.yml

```diff
@@ -1,82 +1,84 @@
+run-name: "Workflow CI/CD Step: cargo build"
 
 on:
   workflow_call:
-    inputs:
-      CACHING_MODE:
-        required: true
-        type: string
-        
-env:
-  IMAGE_NAME: zencash/sc-ci-base
-  IMAGE_TAG: noble_rust-stable_latest
-  DOCKER_COMPOSE_CMD: "docker compose -f ${GITHUB_WORKSPACE}/ci/docker-compose.yml run --rm cargo-container"
-  DOCKER_BUILD_DIR: /build
-  DOCKER_CARGO_HOME: /tmp/.cargo
-  CMAKE_INSTALL: true
-  LLD_INSTALL: true
+  workflow_dispatch:
 
 jobs:
-  build:
-    runs-on:
-      group: ${{ vars.RUNNER_GROUP }}
+  build-mainchain:
+    runs-on: warp-ubuntu-latest-x64-8x
+    name: Build mainchain
     steps:
-      - name: Git checkout
+      - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Set up deps cache
-        run: mkdir deps
-
-      - name: Restore main cache
-        if: ${{ !env.ACT && inputs.CACHING_MODE == 'CACHE_RESTORE' }}
-        uses: buildjet/cache/restore@v3
+      - name: Solo chain build
+        uses: ./.github/actions/cargo-cmd
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-            deps/
-          key: build-cache
+          command: "cargo build -p mainchain --release"
+          use_cache: 'yes'
+          cache_key: 'build-mainchain'
+          cmake_install: 'yes'
+          lld_install: 'yes'
 
-      - name: Set docker env vars
-        run: |
-          echo "USER_ID=$(id -u)" >> $GITHUB_ENV
-          echo "GRP_ID=$(id -g)" >> $GITHUB_ENV
+      - name: Upload zkv-node binary artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: "zkv-node"
+          path: target/release/zkv-node
+          retention-days: 1
+          overwrite: true
 
-      - name: Cargo build solo chain
-        run: ${{ env.DOCKER_COMPOSE_CMD }} cargo build -p mainchain --release
+  build-relay-paratest:
+    runs-on: warp-ubuntu-latest-x64-8x
+    name: Build relay chain and paratest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-      - name: Upload solochain runtime artifact
-        uses: actions/upload-artifact@v4
+      - name: Relay chain and paratest build
+        uses: ./.github/actions/cargo-cmd
         with:
-          name: "zkv_runtime.solochain.compact.compressed.wasm"
-          path: target/release/wbuild/zkv-runtime/zkv_runtime.compact.compressed.wasm
-      
-      - name: Cargo build relay chain
-        run: ${{ env.DOCKER_COMPOSE_CMD }} cargo build -p zkv-relay --release
+          command: "cargo build -p zkv-relay -p paratest-node --release --features fast-runtime"
+          use_cache: 'yes'
+          cache_key: 'build-relay-paratest'
+          cmake_install: 'yes'
+          lld_install: 'yes'
 
-      - name: Upload relay chain runtime artifact
+      - name: Upload relay and paratest build binaries
```

Comment:
Same comment as before, too big to upload them ?

Reply:
We need them for zombienet. Don't it?

Reply:
Of course, but why uploading them ? Can't we save them locally somewhere ?

Reply:
Same answer as before. It is ok to upload them as artifacts, stored for 1 day with default compression level 6

![image](https://github.com/user-attachments/assets/532b24ac-d0bb-4d5a-9391-daeacb38a733)